### ISSUE TITLE
Integration Psalm Security Scan Actions in a Security Starter Workflow

### ISSUE THREAD

**Issue by eroullit** [psalm/psalm-github-security-scan#6]
Hi Psalm Team ! 👋 

GitHub offers a wide variety of [starter workflows](https://docs.github.com/en/actions/using-workflows/using-starter-workflows) to help the community integrate new CI pipelines quickly and efficiently.

I have prepared a [pull request](https://github.com/actions/starter-workflows/pull/1918) aiming to integrate Psalm Security Scan workflow within it.
To finalise it, your action would need to be registered in GitHub [technology partner program](https://partner.github.com/apply?partnershipType=Technology+Partner)

If you are interested, could you fill out this form ?

In any case, feel free to contact me.

**Comment by weirdan** [comment]
I suppose we need to fix the build first.

**Comment by eroullit** [comment]
Hey @weirdan ! 👋 

The build is back in shape and we start getting positive feedback about it ✨ 

I would like to finalise the work initiated on the starter-workflow repo to make it a breeze to integrate Psalm Security Scan workflow in a repo.

For that, the `psalm/psalm-github-security-scan` action would need to be enrolled in the GitHub [technology partner program](https://partner.github.com/apply?partnershipType=Technology+Partner).

**Comment by weirdan** [comment]
That form doesn't make much sense to me. We're not a business (none of the active maintainers, for sure). Vimeo likely holds the copyright (@muglug can you clarify?) but is not actively involved in Psalm anymore, as far as I can tell.

**Comment by eroullit** [comment]
Hey @weirdan ! 👋 

The [technical partner program](https://partner.github.com/technology-partners) is mainly for third components which are integrated in GitHub itself or in its Marketplace..

Its related terms and conditions can be found [here](https://partner.github.com/technology-partners-terms) and they would need to be agreed by the current maintainers.

**Comment by muglug** [comment]
Hey! The main worry I have here is the increased burden on Psalm’s volunteer maintainers when someone starts using security analysis via this workflow, having not used Psalm before.

Psalm’s security analysis works best in the hands of a security researcher who understands the capabilities of it and similar tools. I’m concerned that push-button workflows that lack a contractual commitment might create an onerous amount of work.

**Comment by eroullit** [comment]
First and foremost, I would like to thanks you for maintain Psalm and all related actions. ❤️ 

It is already helping greatly all PHP-based open source software developers and security researchers to find defects on already existing codebase but also on newly developed code.

User’s feedback about alerts raised by Psalm has been very positive overall.
Even more so since the Psalm security action has been [updated](https://github.com/psalm/psalm-github-security-scan/issues/9).

Many open-source projects backed either by companies or volunteers have chosen to have dedicated starter workflows to give users an easy and correct way to setup the first version of their static analysis workflows such as:

- [Sobelow to scan code written in Elixir for the Phoenix Framework](https://github.com/actions/starter-workflows/blob/main/code-scanning/sobelow.yml)
- [Kubesec to check Helm chart deployment on Kubernetes clusters](https://github.com/actions/starter-workflows/blob/main/code-scanning/kubesec.yml)
- [Clippy to check for common programming mistakes in Rust](https://github.com/actions/starter-workflows/blob/main/code-scanning/rust-clippy.yml)
- [Rubocop for Ruby](https://github.com/actions/starter-workflows/blob/main/code-scanning/rubocop.yml)

So far, the synergies between the security researchers building these tools and the developers have been a boon for software security allowing them to ship safer and better code overall.

**Comment by griffinashe** [comment]
Hi @muglug, I work on our security ecosystem team here at GitHub. Would you be open to talking through this some time soon?

**Comment by orklah** [comment]
Hey @griffinashe! Matt no longer works on Psalm. I could try to answer your questions though I'm not exactly sure to understand all the ins and out of what is suggested in this thread

**Comment by griffinashe** [comment]
@orklah - Sorry for the delayed response. If you email me @griffinashe@github.com with some times that work for you the week of July 10th or July 17th I can send an invite out to discuss.

**Comment by orklah** [comment]
I'm not very used to have oral discussion in English. Could that be an email or a chat?

**Comment by griffinashe** [comment]
@orklah Of course. Please send me an email at the address in my previous message and we can discuss there.
